src/app/shared/components/upsert-address/upsert-address.component.ts

```diff
@@ -11,105 +12,134 @@ import {HttpClient} from '@angular/common/http';
-    if (this.data.addressId) {
-      this.addressData.recipient_name = tempName;
-      this.addressData.recipient_mobile_no = this.addressForm.controls['phoneNumber'].value;
-      this.addressData.recipient_national_id = this.addressForm.controls['nationalCode'].value;
+    if (this.addressInfo.addressId) {
+      Object.assign(this.addressData, {
+        recipient_name: tempName,
+        recipient_mobile_no: this.addressForm.controls['phoneNumber'].value,
+        recipient_national_id: this.addressForm.controls['nationalCode'].value,
+      });
     } else {
-      this.addressData.recipient_name = tempName;
-      this.addressData.recipient_mobile_no = this.addressForm.controls['phoneNumber'].value;
-      this.addressData.recipient_national_id = this.addressForm.controls['nationalCode'].value;
-      this.addressData.ostan = this.addressForm.controls['selectOstan'].value;
-      this.addressData.city = this.addressForm.controls['selectCity'].value;
-      this.addressData.street = this.addressForm.controls['street'].value;
-      this.addressData.no = this.addressForm.controls['pelak'].value;
-      this.addressData.unit = this.addressForm.controls['unit'].value;
-      this.addressData.postal_code = this.addressForm.controls['postal_code'].value;
-      this.addressData.distinct = this.addressForm.controls['district'].value;
-      this.addressData.loc = {
-        long: this.addressForm.controls['longitude'].value,
-        lat: this.addressForm.controls['latitude'].value,
-      };
+      Object.assign(this.addressData, {
+        recipient_name: tempName,
+        recipient_mobile_no: this.addressForm.controls['phoneNumber'].value,
+        recipient_national_id: this.addressForm.controls['nationalCode'].value,
+        province: this.addressForm.controls['selectProvince'].value,
+        city: this.addressForm.controls['selectCity'].value,
+        street: this.addressForm.controls['street'].value,
+        no: this.addressForm.controls['pelak'].value,
+        unit: this.addressForm.controls['unit'].value,
+        postal_code: this.addressForm.controls['postal_code'].value,
+        distinct: this.addressForm.controls['district'].value,
+        loc: {
+          long: this.addressForm.controls['longitude'].value,
+          lat: this.addressForm.controls['latitude'].value,
+        }
+      });
     }
-    this.dialogRef.close(this.addressData);
+    // console.log("final address data are : ", this.addressData);
+    this.checkoutService.submitAddresses(this.addressData)
+      .then(res => {
+        this.onClose();
+      })
+      .catch(err => {
+        console.error('error occured in submitting address', err);
+      });
   }
 
-  setNewOstan(newOstan) {
-    this.cityArray = this.ostanArray.find(el => el.name === newOstan).cities;
-    this.addressData.ostan = newOstan;
+  setNewProvince(newProvince) {
+    this.cityArray = this.provinceArray.find(el => el.name === newProvince).cities;
```

Comment:
Highly recommended to convert both el.name and newProvince to lower case then compare them